semgrep-core/src/engine/Eval_generic.ml

```diff
@@ -295,12 +295,18 @@ let bindings_to_env xs =
 (* this is for metavariable-regexp *)
 let bindings_to_env_with_just_strings xs =
   xs
-  |> List.map (fun (mvar, mval) ->
+  |> Common.map_filter (fun (mvar, mval) ->
          let any = MV.mvalue_to_any mval in
-         let min, max = Visitor_AST.range_of_any any in
-         let file = min.Parse_info.file in
-         let range = Range.range_of_token_locations min max in
-         (mvar, String (Range.content_at_range file range)))
+         match Visitor_AST.range_of_any_opt any with
+         | None ->
```

Comment:
Can we factorize the error reporting? Maybe make a `no_range_info_error` function, and if we later decide we want to report a warning to the user we can add it there.